src/main/webapp/scripts/map.js

```diff
@@ -187,20 +187,19 @@ function placeMarkerAndPanTo(latLng, map) {
 }
 
 /* Function to place markers from the datastore */
-function placeMarker(latLng, map) {
+function placeMarker(markerDetails, map) {
   const marker = new google.maps.Marker ({
-    position: latLng,
+    position: new google.maps.LatLng(markerDetails.lat, markerDetails.lng),
     map: map,
     icon: greyIcon, // TODO change the colour of the icon depending on the category
   });
+  marker.set("id", markerDetails.id);
```

Comment:
are we going to add the "click" eventlistener to the marker for listing the recommendation here?
if yes I was thinking maybe, instead of setting the id,  we can just put the id into the fucntion, e.g. google.maps.event.addListener(marker, "click", () => listRecommendation(markerDetails.id));
listingRecommendation function will fetch the recommendation details from datastore using the Id then proceed to display the details ( line 126-134)
then we can remove addHardCodedMarkers  function

Reply:
Yes, there will be an event listener added but that is not at this iteration of the code - this is just to store and then place the markers (rather than listing).